src/main/java/com/alextherapeutics/diga/implementation/DigaXmlJaxbRequestReader.java

```diff
@@ -59,36 +59,45 @@ public DigaXmlJaxbRequestReader() throws JAXBException {
+            return DigaInvoiceResponse.builder()
+                    .hasError(!report.isValid())
+                    .errors(getInvoiceErrors(report))
+                    .rawXmlResponseBody(bytes)
+                    .build();
+        } catch (JAXBException | IOException e) {
+            throw new DigaXmlReaderException(e);
+        }
 
     }
 
     @Override
-    public DigaCodeValidationResponse readCodeValidationResponse(InputStream decryptedResponse) throws JAXBException, IOException {
-        var bytes = decryptedResponse.readAllBytes();
-        var response = (PruefungFreischaltcode) codeValidationUnmarshaller.unmarshal(new ByteArrayInputStream(bytes));
-        validateCodeValidationResponse(response);
-
-        // appendix 4 at https://www.gkv-datenaustausch.de/leistungserbringer/digitale_gesundheitsanwendungen/digitale_gesundheitsanwendungen.jsp
-        // seems to differ from the current xsd schema definition. watch for changes here
-        return DigaCodeValidationResponse.builder()
-                .rawXmlResponseBody(IOUtils.toString(bytes, "UTF-8"))
-                .hasError(response.getNachrichtentyp().equals(NachrichtentypStp.FEH))
-                .errors(getCodeValidationErrors(response))
-                .validatedDigaCode(response.getAntwort() == null ? null : response.getAntwort().getFreischaltcode())
-                .dayOfServiceProvision(response.getAntwort() == null ? null : response.getAntwort().getTagDerLeistungserbringung().toGregorianCalendar().getTime())
-                .validatedDigaveid(response.getAntwort() == null ? null : response.getAntwort().getDiGAVEID())
-                .build();
+    public DigaCodeValidationResponse readCodeValidationResponse(InputStream decryptedResponse) throws DigaXmlReaderException {
+        try {
+
+            var bytes = decryptedResponse.readAllBytes();
+            var response = (PruefungFreischaltcode) codeValidationUnmarshaller.unmarshal(new ByteArrayInputStream(bytes));
+            validateCodeValidationResponse(response);
+
+            // appendix 4 at https://www.gkv-datenaustausch.de/leistungserbringer/digitale_gesundheitsanwendungen/digitale_gesundheitsanwendungen.jsp
+            // seems to differ from the current xsd schema definition. watch for changes here
+            return DigaCodeValidationResponse.builder()
+                    .rawXmlResponseBody(bytes)
+                    .hasError(response.getNachrichtentyp().equals(NachrichtentypStp.FEH))
+                    .errors(getCodeValidationErrors(response))
+                    .validatedDigaCode(response.getAntwort() == null ? null : response.getAntwort().getFreischaltcode())
+                    .dayOfServiceProvision(response.getAntwort() == null ? null : response.getAntwort().getTagDerLeistungserbringung().toGregorianCalendar().getTime())
+                    .validatedDigaveid(response.getAntwort() == null ? null : response.getAntwort().getDiGAVEID())
```

Comment:
Can you do "||" here instead of "null ? null : ..."?

Reply:
nope! it is not possible in Java :)